src/lib.rs

```diff
@@ -1,18 +1,146 @@
 use bitcoin::blockdata::transaction::{TxOut, TxIn, Transaction};
-use secp256k1::PublicKey;
+use bitcoin::blockdata::script::{Script, Builder};
+use bitcoin::blockdata::opcodes;
+use bitcoin::util::address::Address;
+use secp256k1::key::{PublicKey};
+use bitcoin::hashes::{Hash, sha256d};
 
-use lightning::ln::chan_utils::*;
+use secp256k1::*;
 
 const DUST_LIMIT: u64 = 5000;
 const TX_VERSION: u32 = 2;
+const MATURITY_TIME_MIN: u32 = 500000000;
+
+pub fn combine_keys(pub_keys: &[PublicKey]) -> PublicKey {
+    pub_keys.iter().fold(pub_keys[0], |keys, key| keys.combine(key).unwrap())
+}
+
+pub fn get_secp_committed_key(oracle_pub_key: PublicKey,
+                              oracle_r_point: PublicKey,
+                              message: String) -> PublicKey {
+    let s = Secp256k1::signing_only();
+    let msg = Message::from_slice(message.as_bytes()).unwrap();
```

Comment:
IIRC this needs to be 32 bytes so you need to hash it first.

Reply:
Thanks for catching this.  I'll update it.